pymc3/step_methods/pgbart.py

```diff
@@ -0,0 +1,263 @@
+#   Copyright 2020 The PyMC Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import numpy as np
+from theano import function as theano_function
+
+from .arraystep import ArrayStepShared, Competence
+from ..distributions import BART
+from ..distributions.tree import Tree
+from ..model import modelcontext
+from ..theanof import inputvars, make_shared_replacements, join_nonshared_inputs
+
+
+class PGBART(ArrayStepShared):
+    """
+    Particle Gibss BART sampling step
+
+    Parameters
+    ----------
+    vars: list
+        List of variables for sampler
+    num_particles : int
+        Number of particles for the SMC sampler. Defaults to 10
```

Comment:
SMC -> PGBART, same below

Reply:
I tried to follow the nomenclature in the papers and one step of the PGBART is "conditional-SMC" method. But I see how this can be confusing.